pkg/executor/aggfuncs/spill_serialize_helper.go

```diff
@@ -0,0 +1,255 @@
+	return s.buf[0:end]
+}
+
+func (s *spillSerializeHelper) serializePartialResult4MaxMinTime(value partialResult4MaxMinTime) []byte {
+	*(*types.Time)(unsafe.Pointer(&s.buf[0])) = value.val
+	end := timeLen + boolLen
+	spill.SerializeBool(value.isNull, s.buf[timeLen:end])
+	return s.buf[0:end]
+}
+
+func (s *spillSerializeHelper) serializePartialResult4MaxMinDuration(value partialResult4MaxMinDuration) []byte {
+	spill.SerializeInt64(int64(value.val.Duration), s.buf[0:int64Len])
+	spill.SerializeInt(value.val.Fsp, s.buf[int64Len:int64Len+intLen])
+	end := int64Len + intLen + boolLen
+	spill.SerializeBool(value.isNull, s.buf[int64Len+intLen:end])
+	return s.buf[0:end]
+}
+
+func (s *spillSerializeHelper) serializePartialResult4MaxMinString(value partialResult4MaxMinString) []byte {
+	spill.SerializeBool(value.isNull, s.buf[0:boolLen])
+	s.buf = s.buf[:boolLen]
+	s.buf = append(s.buf, value.val...)
+	return s.buf
+}
+
+func (s *spillSerializeHelper) serializePartialResult4MaxMinJSON(value partialResult4MaxMinJSON) []byte {
+	spill.SerializeBool(value.isNull, s.buf[0:])
+
+	// Assign the return value to `s.buf` is necessary, so that the `s.buf` could expand it's capacity.
+	s.buf = spill.SerializeBinaryJSON(&value.val, s.buf, boolLen)
+	return s.buf
+}
+
+func (s *spillSerializeHelper) serializePartialResult4MaxMinEnum(value partialResult4MaxMinEnum) []byte {
+	spill.SerializeBool(value.isNull, s.buf[0:])
+
+	// Assign the return value to `s.buf` is necessary, so that the `s.buf` could expand it's capacity.
+	s.buf = spill.SerializeEnum(&value.val, s.buf, boolLen)
+	return s.buf
+}
+
+func (s *spillSerializeHelper) serializePartialResult4MaxMinSet(value partialResult4MaxMinSet) []byte {
+	spill.SerializeBool(value.isNull, s.buf)
+	s.buf = spill.SerializeSet(&value.val, s.buf, boolLen)
+	return s.buf
+}
+
+func (s *spillSerializeHelper) serializePartialResult4AvgDecimal(value partialResult4AvgDecimal) []byte {
+	*(*types.MyDecimal)(unsafe.Pointer(&s.buf[0])) = value.sum
+	spill.SerializeInt64(value.count, s.buf[types.MyDecimalStructSize:])
+	return s.buf[0 : types.MyDecimalStructSize+int64Len]
+}
+
+func (s *spillSerializeHelper) serializePartialResult4AvgFloat64(value partialResult4AvgFloat64) []byte {
+	spill.SerializeFloat64(value.sum, s.buf[:])
+	spill.SerializeInt64(value.count, s.buf[float64Len:])
+	return s.buf[0 : float64Len+int64Len]
+}
+
+func (s *spillSerializeHelper) serializePartialResult4SumDecimal(value partialResult4SumDecimal) []byte {
+	*(*types.MyDecimal)(unsafe.Pointer(&s.buf[0])) = value.val
+	spill.SerializeInt64(value.notNullRowCount, s.buf[types.MyDecimalStructSize:])
+	return s.buf[0 : types.MyDecimalStructSize+int64Len]
+}
+
+func (s *spillSerializeHelper) serializePartialResult4SumFloat64(value partialResult4SumFloat64) []byte {
+	spill.SerializeFloat64(value.val, s.buf[:])
+	spill.SerializeInt64(value.notNullRowCount, s.buf[float64Len:])
+	return s.buf[0 : float64Len+int64Len]
```

Comment:
why not `return s.buf` directly as `serializeBasePartialResult4GroupConcat` does , `return s.buf[0 : float64Len+int64Len]` makes the code a little hard to read

Reply:
> why not `return s.buf` directly as `serializeBasePartialResult4GroupConcat` does , `return s.buf[0 : float64Len+int64Len]` makes the code a little hard to read

Because we can't guarantee that the size of buf is just the same with the size of `partialResult4SumFloat64`, but I think we can define a const value that represents the size of `partialResult4SumFloat64` in advance so that it's more readable.